Add ingame player name to party data

TheStonedTurtle opened the pull request:
Adds a config option to show player name in overlays instead of discord name. Will show the other on hover.

![https://i.imgur.com/Sk7a71m.png](https://i.imgur.com/Sk7a71m.png)
![https://i.imgur.com/p3vrPnm.png](https://i.imgur.com/p3vrPnm.png)
![https://i.imgur.com/9UP9gUx.png](https://i.imgur.com/9UP9gUx.png)

forsco commented:
A note: I couldn't see `L atios` in the party in my client when Stoned took this screenshot.

TheStonedTurtle commented:
Not sure if the issue Forsco ran into could be because of the new event being posted not being registered on the client he was running.

deathbeam commented:
This removes information about who is the person on Discord. I don't think that is good thing. I would rather extend player indicators to draw discord name on top of party member.

forsco commented:
How many players do you think who will end up using this will want to have discord names on the overlay instead of OSRS names? seems counter-intuitive for an in-game display.

deathbeam commented:
Well I would still like to know who on Discord I am actually playing with.

forsco commented:
You would know when you ask to join the group, or when someone asks to join yours. Okay, I have `BobJohn#1111` in my party but in game name is completely different. As an example you want to cast heal other BobJohn but now also have to remember their in game name rather than just displaying the name that matters whilst inside the game.

If displaying the discord name is such a big deal to you, I guess the option of choice could be offered for what is displayed as the name?

deathbeam commented:
You would not know. You are not always owner, and also that would mean you would like need to watch changes in party overlay in-game. 

I dont disagree that mapping the discord party member to in-game user is important, but doing the opposite is important too. You can be e.g talking or messaging with your party member in Discord, but then you would not be able to easily assign to who are you talking to in-game.

deathbeam commented:
I think my player indicators solution is best one, because that adds nice and easily understandable mapping between  both worlds. What you are suggesting is not solution because then you have option to either display 1 information or other, so you still dont have the mapping.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/party/PartyPlugin.java:
```diff
@@ -388,6 +404,10 @@ public void onUserSync(final UserSync event)
 			prayUpdate.setMemberId(localMember.getMemberId());
 			ws.send(prayUpdate);
+
+			final UsernameUpdate usernameUpdate = new UsernameUpdate(client.getLocalPlayer().getName());
```
This will NPE if person is not in-game

TheStonedTurtle commented:
> I think my player indicators solution is best one

Should I just close this PR if your solution will accomplish the same task?

TheStonedTurtle commented:
PR has been updated.

1) Renamed username to playerName
2) Only tries to sync your playerName when logged in and now syncs upon login
3) Added a config option for updating the overlay names & added a tooltip to display the other name.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/overlay/components/PanelComponent.java:
```diff
@@ -111,6 +117,8 @@ public Dimension render(Graphics2D graphics)
 			preferredSize.width - border.x - border.width,
 			preferredSize.height - border.y - border.height);
 
+		final Point boundsLoc = new Point(parentBoundsLocation.x + preferredLocation.x, parentBoundsLocation.y + preferredLocation.y);
```
I dont like this parentBounds thing. You can just do that by yourself outside of panel. Or if you really want to implement this properly, add getBounds() to LayoutableRenderableEntity and implement it in other implementations too.

deathbeam replied:
See https://github.com/runelite/runelite/pull/7705

TheStonedTurtle commented:
PR has been updated to include #7705

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/party/PartyPlugin.java:
```diff
@@ -512,4 +553,42 @@ public void keyReleased(KeyEvent keyEvent)
+
+	private void fetchAndSyncPlayerName()
+	{
+		clientThread.invoke(() ->
```
I dont think you need to do this on clientThread

TheStonedTurtle replied:
I invoke this on client thread so that it keeps running until name returns something (or one of the other states happens). Sometimes the name will return null even when logged in.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/party/PartyPlugin.java:
```diff
@@ -413,6 +443,15 @@ public void onUserSync(final UserSync event)
 			prayUpdate.setMemberId(localMember.getMemberId());
 			ws.send(prayUpdate);
+
+			switch (client.getGameState())
```
You can just check localPlayer for being non-null in the method instead of doing game state checks

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/party/PartyPlugin.java:
```diff
@@ -323,6 +331,15 @@ public void onGameTick(final GameTick event)
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged c)
+	{
+		if (c.getGameState().equals(GameState.LOGGING_IN))
```
You can just set boolean here and check it in gameTick

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/party/PartyStatsOverlay.java:
```diff
@@ -134,6 +151,16 @@ public Dimension render(Graphics2D graphics)
+				final Point mousePosition = client.getMouseCanvasPosition();
+				// Adjust bounds to account for overlay position
+				final Rectangle panelBounds = new Rectangle(panel.getBounds());
+				panelBounds.x += getBounds().x;
```
use panelBounds.translate

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/party/PartyStatsOverlay.java:
```diff
@@ -134,6 +151,16 @@ public Dimension render(Graphics2D graphics)
+				final Rectangle panelBounds = new Rectangle(panel.getBounds());
+				panelBounds.x += getBounds().x;
+				panelBounds.y += getBounds().y;
+				if (tooltipText != null && panelBounds.contains(mousePosition.getX(), mousePosition.getY()))
```
add empty line above

TheStonedTurtle commented:
The requested changes have been made to the necessary commits

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/party/PartyPlugin.java:
```diff
@@ -120,14 +123,18 @@
 	private ChatMessageManager chatMessageManager;
 
+	@Inject
+	private ClientThread clientThread;
```
This is unused.

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/party/PartyPlugin.java:
```diff
@@ -317,12 +326,38 @@ public void onGameTick(final GameTick event)
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged c)
+	{
+		if (c.getGameState().equals(GameState.LOGGING_IN))
```
```suggestion
		if (c.getGameState() == GameState.LOGGING_IN)
```

TheStonedTurtle commented:
Nightfire's requested changes have been made

TheStonedTurtle commented:
PR has been updated since #7705 was merged
